website/content/docs/integrations/vault-integration.mdx

```diff
@@ -22,134 +22,356 @@ install Vault separately from Nomad. Nomad does not run Vault for you.
 
 ## Agent Configuration
 
-To enable Vault integration, please see the [Nomad agent Vault
-integration][config] configuration.
+To enable Vault integration, Nomad agents must be configured with
+[`vault`][config] blocks. Each block can be enabled or disabled and configures
+how Nomad connects to the Vault cluster.
 
-## Vault Definition Syntax
+## Vault ACL
 
-To configure a job to retrieve Vault tokens, please see the [`vault` job
-specification documentation][vault-spec].
+The Vault ACL system protects the cluster from unauthorized access. When
+enabled, both Vault and Nomad must be properly configured in order for their
+integrations to work.
 
-## Using Nomad Workload Identity with Vault
+### Nomad Workload Identities
 
 Starting in Nomad 1.7, Nomad clients will use a task's [Workload Identity][] to
 authenticate to Vault and obtain a Vault token specific to the task.
 
-### Configuring Vault Authentication
+By default, Nomad only generates a workload identity for tasks that is used to
+access Nomad itself. To access Vault, jobs must define additional workload
```

Comment:
This line is a bit hard to follow, Nomad only generates identities for workloads that need to access some other Nomad resources?

Reply:
Yup! For example, when reading [Nomad variables from a `template`](https://developer.hashicorp.com/nomad/docs/job-specification/template#nomad-integration) the task needs to access the Nomad API, so it uses this identity for authentication.

Reply:
Expanded to this:

> By default, Nomad does not generate workload identities for services, and tasks only receive an identity that is used to access data from Nomad itself, such as for reading [Variables](https://nomad-git-docs-consul-vault-integrations-hashicorp.vercel.app/nomad/docs/concepts/variables) from a [template](https://nomad-git-docs-consul-vault-integrations-hashicorp.vercel.app/nomad/docs/job-specification/template) block.